validator.js

```diff
@@ -141,6 +141,23 @@ var validator = function(options) {
 
   // Load schemas from folder
   if (options.folder) {
+
+    // Load JSON schema for JSON Schemas
+    var coreSchema = fs.readFileSync(
+      path.join(__dirname, 'schemas', 'draft-04-schema.json'), {
+      encoding: 'utf-8'
+    });
+    // Create validator for core schema, so we can report schemas that are
+    // invalid
+    var coreSchemaValidator = new jjv();
+    coreSchemaValidator.addSchema(JSON.parse(coreSchema));
+    var validateSchema = function(schemaToValidate) {
+      return coreSchemaValidator.validate(
+        'http://json-schema.org/draft-04/schema#',
+        schemaToValidate
+      );
```

Comment:
Have you checked that this doesn't go out on the network to download the schema, and uses the `coreSchema` added to the validator? I'm sure it must, but just thought I'd double check.